src/login/logind-action.c

```diff
@@ -299,12 +300,41 @@ static int handle_action_sleep_execute(
         return handle_action_execute(m, handle, ignore_inhibited, is_edge);
 }
 
+static int manager_handle_action_secure_attention_key(
+                Manager *m,
+                bool is_edge,
+                char *seat) {
+        int r;
+        Seat *o;
+        char *p;
+
+        if (!is_edge)
+                return 0;
+
+        o = hashmap_get(m->seats, seat);
+        p = seat_bus_path(o);
+
+        log_struct(LOG_INFO,
+                   LOG_MESSAGE("Secure Attention Key sequence pressed on seat %s", seat),
+                   "MESSAGE_ID=" SD_MESSAGE_SECURE_ATTENTION_KEY_PRESS_STR);
+
+        r = sd_bus_emit_signal(m->bus,
+                                  "/org/freedesktop/login1",
+                                  "org.freedesktop.login1.Manager",
+                                  "SecureAttentionKey",
+                                  "so", seat, p);
+
+        free(p);
```

Comment:
use `_cleanup_free_` when you declare the thing.

Reply:
Do I remove this free(); when I do so?